### PR TITLE
bypass cache for demo ui site build.

### PR DESCRIPTION
**Pull request by romaninsh**
This adds a new step into "atk-demo" build pipeline, which would specify a commit to be used by dependencies.

This helps us to avoid packagist cache and add "triggers" for dependency libraries too

Screenshot 2020-06-02 at 18 41 54

Screenshot_2020-06-02_at_18 17 09

### CODE REVIEW

File: tools/inject-dependency-sha.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+cat composer.json | \
+  jq '.require["atk4/core"]=.require["atk4/core"]+"#'`git ls-remote git@github.com:atk4/core.git develop | cut -f1`'"' |\
+  jq '.require["atk4/data"]=.require["atk4/data"]+"#'`git ls-remote git@github.com:atk4/data.git develop | cut -f1`'"' \
```
Comment (mvorisek) [inline tools/inject-dependency-sha.sh]: Add dsql as well

**Review by mvorisek — CHANGES_REQUESTED** [review]

File: .codefresh/ui-build-deploy.yaml [hunk 2]
```diff
@@ -10,6 +10,13 @@ steps:
     stage: clone
     repo: atk4/ui
     revision: "${{CF_BRANCH}}"
+
+  fix_sha:
+    stage: clone
+    title: Inject dependency SHA to bypass cache
+    commands:
+      - ./tools/inject-dependency-sha.sh
```
Comment (mvorisek) [inline .codefresh/ui-build-deploy.yaml]: I think it is still better to do in DockerFile, as it can be run also locally.

**Review by mvorisek — CHANGES_REQUESTED** [review]

File: tools/inject-dependency-sha.sh (hunk 1 above)
Comment (romaninsh) [reply tools/inject-dependency-sha.sh]: dsql is not a direct dependency of atk4/ui, i'll try adding it though.

File: .codefresh/ui-build-deploy.yaml (hunk 2 above)
Comment (romaninsh) [reply .codefresh/ui-build-deploy.yaml]: if files are produced on a separate step, it allows us to cache docker layers more efficiently when dependencies haven't been updated.

Comment (romaninsh) [reply .codefresh/ui-build-deploy.yaml]: just thought of another reason for this being in the pipeline...

We are likely to have 2 pipelines - one for stable builds (e.g. master) and another for unstable (develop).

File: tools/inject-dependency-sha.sh (hunk 1 above)
Comment (romaninsh) [reply tools/inject-dependency-sha.sh]: actually - can we divert the 'dsql' dependency addition into 2.2 ?

**Review by DarkSide666 — APPROVED** [review]
LGTM

File: .codefresh/ui-build-deploy.yaml (hunk 2 above)
Comment (mvorisek) [reply .codefresh/ui-build-deploy.yaml]: Really? composer install can not be cached, so if this is after it...

**Pull request merged**